### PR TITLE
Issue300 fix debug spectrograms

### PR DESCRIPTION
**Pull request by towsey**
Original intention was to fix drawing of spectrograms. This included the drawing of acoustic events and the hit matrix overlap. Therefore did some tidying up of the AcousticEvent class. Also set FFT window to HANNING which was originally set as Issue #291.
Wrote unit tests for the drawing of Acoustic events and the merging of overlapping acoustic events.

Work for #300.

### CODE REVIEW

**Comment by towsey** [comment]
"hit matrix overlap" should read "hit matrix overlay"

File: tests/Acoustics.Test/AudioAnalysisTools/AcousticEvents/EventTests.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            Assert.AreEqual(8000, events[1].HighFrequencyHertz);
+
+            Assert.AreEqual(5.0, events[0].Score, 1E-4);
+            Assert.AreEqual(9.0, events[1].Score, 1E-4);
+            Assert.AreEqual(0.5, events[0].ScoreNormalised, 1E-4);
+            Assert.AreEqual(0.9, events[1].ScoreNormalised, 1E-4);
+        }
+
+        [TestMethod]
+        public void TestSonogramWithEventsOverlay()
```
Comment (atruskie) [inline tests/Acoustics.Test/AudioAnalysisTools/AcousticEvents/EventTests.cs]: For this test, I'd suggest using the `GeneratedImageTest` base class I've set up.

Also, the previous comment about ceremony applies here

File: tests/Acoustics.Test/AudioAnalysisTools/AcousticEvents/EventTests.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            // now add events into the spectrogram image.
+            // set colour for the events
+            foreach (AcousticEvent ev in events)
+            {
+                ev.BorderColour = AcousticEvent.DefaultBorderColor;
+                ev.ScoreColour = AcousticEvent.DefaultScoreColor;
+                ev.DrawEvent(substituteSonogram, framesPerSecond, freqBinWidth, height);
+            }
+
+            //substituteSonogram.Save("C:\\temp\\image.png");
```
Comment (atruskie) [inline tests/Acoustics.Test/AudioAnalysisTools/AcousticEvents/EventTests.cs]: This is a bad example to leave in. Remove or fix path. See `GeneratedImageTests` for example output save directory

File: tests/Acoustics.Test/AudioAnalysisTools/StandardSpectrograms/SonogramTests.cs [hunk 3]
```diff
@@ -219,5 +221,47 @@ public void TestAnnotatedSonogramWithPlots()
             Assert.AreEqual(1621, image.Width);
             Assert.AreEqual(656, image.Height);
         }
+
+        [TestMethod]
+        public void TestSonogramHitsOverlay()
```
Comment (atruskie) [inline tests/Acoustics.Test/AudioAnalysisTools/StandardSpectrograms/SonogramTests.cs]: Use the generated image test class

**Comment by towsey** [comment]
Concerning requested changes:
1: I have reverted the names of temporal variables as requested.
2: Changed signatures of test methods to simplify.
3: Removed methods that were commented out.
4: I have not used the GeneratedImageTest base class because I do not understand how to make use of the examples made available.
5: I created a new name space for Event tests, because it seems sensible to have all test classes of events in the one (more general) name space.

**Comment by atruskie** [comment]
@towsey - thanks.

RE 4.: Set up a call.
RE 5. Tests should *roughly* mirror source. The event statistics class is in `AudioAnalysisTools.EventStatistics` in the source so the test should be in `Acoustics.Test.AudioAnalysisTools.EventStatistics`. What made sense to you is actually less consistent.

**Comment by towsey** [comment]
@atruskie 
Concerning point 5, location of the EventStatistics class, my suggestion is that the source should be altered so that both the EventBase class, the AcousticEvent class and the EventStatistics class are all included within the same namespace, e.g. Event namespace.

I will fix the Anti Aliasing issue and then suggest a call  concerning point 4.

**Comment by towsey** [comment]
I have not been able to find a solution to the anti-aliasing problem.
Can you send me an example or can we have a call when you are ready?

**Pull request merged**